github_api.module

```diff
@@ -325,3 +325,87 @@ function github_api_delete_branch($username = NULL, $repository = NULL, $source_
     watchdog('github_api', '@exception', array('@exception' => $exception->getMessage()), WATCHDOG_ERROR);
   }
 }
+
+/**
+ * Callback to fetch a repository's latest release.
+ *
+ * @param string $username
+ *   The GitHub username of the repo owner.
+ *
+ * @param string $repository
+ *   The name of the repository.
+ *
+ * @throws Exception
+ */
+function github_api_get_latest_release($username = NULL, $repository = NULL) {
+  if (!$username) {
+    $username = variable_get('github_api_default_owner');
+  }
+
+  $client = github_api_client();
+
+  try {
+    $client->api('current_user')->show();
+    $release = $client->api('repo')->releases()->latest($username, $repository);
+
+    if (!(is_array($release))) {
+      throw new Exception('Fetch latest release failed.');
+    }
+
+    return $release;
+  }
+  catch (Exception $exception) {
+    watchdog('github_api', '@exception', array('@exception' => $exception->getMessage()), WATCHDOG_ERROR);
+  }
+}
+
+/**
+ * Callback to fetch a binary asset by ID and save it as a managed file.
+ *
+ * @param string $username
+ *   The GitHub username of the repo owner.
+ *
+ * @param string $repository
+ *   The name of the repository.
+ *
+ * @param int $asset_id
+ *   The asset's ID.
+ *
+ * @param string $uri
+ *   The URI to which to save the file. (e.g. 'public://filename.zip')
+ *
+ * @return object $file
+ *   Drupal file object
+ *
+ * @throws Exception
+ */
+function github_api_fetch_asset($username = NULL, $repository, $asset_id, $uri) {
+  if (!$username) {
+    $username = variable_get('github_api_default_owner');
+  }
+
+  $client = github_api_client();
+
+  try {
+    $url = url(
```

Comment:
@acrollet It's great to see you adding more functionality to the library.

The idea of this module is that it is a lightweight wrapper around [https://github.com/KnpLabs/php-github-api](KnP Lab's github library). They have implemented support for [release assets](https://github.com/KnpLabs/php-github-api/blob/master/lib/Github/Api/Repository/Assets.php). The composer.json should be updated to so the dependency is `^1.5.1` and then instead of using `drupal_http_request()` to make this call it should use the library.

I haven't run coder-review over the patch, but other than the issue above I don't see any other issues with the patch.  If you fix it and squash the commits it should be an easy merge.